### DISCUSSION TITLE
vpn client DNS

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#834] [General]
i've set DNS on my server for clients pools, but client dns different .

**Server Settings:**
![image](https://user-images.githubusercontent.com/17877111/148211547-b83d527c-e560-4108-9a6f-0af4a2b5b124.png)

**Client (Windows) DNS**

![image](https://user-images.githubusercontent.com/17877111/148211607-415dea54-fd2e-4e4c-a4b7-12036db536ed.png)

Should the server give this DNS to the client ?, or did I get it wrong, the other thing is where those values come from (Client DNS) . I know it's Google's DNS, but this DNS not set on any of  server or client pc.

the main idea is forcing my vpn client to use my own DNS to block some bad websites. 

Thanks in advance

**Comment by tobiasbrunner** [comment]
Read the log, you'll see whether the client requested DNS servers and what values the VPN server assigned.

**Reply by realbitt** [reply]
**client log.**
[IKE] Installing DNS Server 8.8.8.8
[IKE] Installing DNS Server 8.8.4.4
[IKE] Installing DNS Server 122.99.12.11
[IKE] Installing DNS Server 1.1.1.1

**Reply by tobiasbrunner** [reply]
Looks like you have the other servers defined somewhere (e.g. for the [attr plugin](https://wiki.strongswan.org/projects/strongswan/wiki/AttrPlugin) in strongswan.conf or an included attr.conf).

**Reply by realbitt** [reply]
I found it in  strongswan.conf thanks @tobiasbrunner
